en/advanced-topics/jsb-auto-binding.md

```diff
@@ -0,0 +1,374 @@
+{
+    auto* cobj = SE_THIS_OBJECT<abc::JSBBridge>(s);
+    SE_PRECONDITION2(cobj, false, "js_cocos2dx_test_JSBBridge_abcLog : Invalid Native Object");
+    const auto& args = s.args();
+    size_t argc = args.size();
+    CC_UNUSED bool ok = true;
+    if (argc == 1) {
+        HolderType<std::string, true> arg0 = {};
+        ok &= sevalue_to_native(args[0], &arg0, s.thisObject());
+        SE_PRECONDITION2(ok, false, "js_cocos2dx_test_JSBBridge_abcLog : Error processing arguments");
+        cobj->abcLog(arg0.value());
+        return true;
+    }
+    SE_REPORT_ERROR("wrong number of arguments: %d, was expecting %d", (int)argc, 1);
+    return false;
+}
+SE_BIND_FUNC(js_cocos2dx_test_JSBBridge_abcLog)
+
+static bool js_cocos2dx_test_JSBBridge_getInstance(se::State& s) // NOLINT(readability-identifier-naming)
+{
+    const auto& args = s.args();
+    size_t argc = args.size();
+    CC_UNUSED bool ok = true;
+    if (argc == 0) {
+        abc::JSBBridge* result = abc::JSBBridge::getInstance();
+        ok &= nativevalue_to_se(result, s.rval(), nullptr /*ctx*/);
+        SE_PRECONDITION2(ok, false, "js_cocos2dx_test_JSBBridge_getInstance : Error processing arguments");
+        SE_HOLD_RETURN_VALUE(result, s.thisObject(), s.rval());
+        return true;
+    }
+    SE_REPORT_ERROR("wrong number of arguments: %d, was expecting %d", (int)argc, 0);
+    return false;
+}
+SE_BIND_FUNC(js_cocos2dx_test_JSBBridge_getInstance)
+
+bool js_register_cocos2dx_test_JSBBridge(se::Object* obj) // NOLINT(readability-identifier-naming)
+{
+    auto* cls = se::Class::create("JSBBridge", obj, nullptr, nullptr);
+
+    cls->defineFunction("abcLog", _SE(js_cocos2dx_test_JSBBridge_abcLog));
+    cls->defineStaticFunction("getInstance", _SE(js_cocos2dx_test_JSBBridge_getInstance));
+    cls->install();
+    JSBClassType::registerClass<abc::JSBBridge>(cls);
+
+    __jsb_abc_JSBBridge_proto = cls->getProto();
+    __jsb_abc_JSBBridge_class = cls;
+
+    se::ScriptEngine::getInstance()->clearException();
+    return true;
+}
+bool register_all_cocos2dx_test(se::Object* obj)
+{
+    // Get the ns
+    se::Value nsVal;
+    if (!obj->getProperty("abc", &nsVal))
+    {
+        se::HandleObject jsobj(se::Object::createPlainObject());
+        nsVal.setObject(jsobj);
+        obj->setProperty("abc", nsVal);
+    }
+    se::Object* ns = nsVal.toObject();
+
+    js_register_cocos2dx_test_JSBBridge(ns);
+    return true;
+}
+```
+
+Doesn't it look familiar? It's exactly the same as Cocos' existing `.cpp` files, even including the registration functions and class definitions that are all automatically generated.
+
+## Cocos Compilation Configuration
+
+Although we have generated the bindings after the above step, the Typescript layer can't be used directly because we still need to configure the generated bindings into the `CMakeLists.txt` file to be compiled with other C++ files, which is the last part of the `CMakeLists.txt` compilation configuration.
```

Comment:
```suggestion
Although we have generated the bindings after the above step, the TypeScript layer can't be used directly because we still need to configure the generated bindings into the `CMakeLists.txt` file to be compiled with other C++ files, which is the last part of the `CMakeLists.txt` compilation configuration.
```